### ISSUE TITLE
Create community-oriented system for showing off tree-sitter parsing rates

### ISSUE THREAD

**Issue by mjambon** [semgrep/semgrep#3897]
The goals are:
1. Make it easy for everyone to see the parse rate (% lines parsed successfully) for each tree-sitter parser that exists, independently from semgrep.
2. Make it easy for the authors of tree-sitter parsers to publish these parsing stats for their project (by providing a list of public projects to scan + list of file extensions or some other mechanism to select target files of the correct type)

The main discussion started here: https://github.com/tree-sitter/tree-sitter/discussions/1001

**Comment by emjin** [comment]
More repos for us to benchmark on!

**Comment by mjambon** [comment]
For storing the data and making graphs, we should probably take advantage of an existing service rather than rolling our own. @underyx once suggested Datadog.

**Comment by aryx** [comment]
I would add 
3. provide a way to see parsing regressions for contributors.

This is super important. I recently had to revert a few PRs on a tree-sitter repo because those PRs were introducing big regressions.

**Comment by mjambon** [comment]
> provide a way to see parsing regressions for contributors.

Yes, this would be nice to have but also it looks like we'd need branch-specific runs (pre-merge), which would work best as GitHub-triggered, per-repo CI rather than a centralized scanner that runs periodically.

There are several advantages to running a centralized system for scanning external repos:
* no need for each project's cooperation.
* less work for each project maintainer.
* creates incentives for tree-sitter projects to follow a standard repo structure: "if you want your project on the public dashboard, you need to stick to the following conventions".

We could also offer a simple mechanism for checking a specific branch of a tree-sitter repo, which would reuse the conventions and the code used by the centralized scanner. The components needed for this are:
* list of public git URLs + commit ID
* target filter (by extension or more abstract like a command that takes a file path and returns 0/1)
* scanning program
It's important that the above change as little as possible. Changes should be easy to trace. To compare two versions of the same grammar, we only want the tree-sitter repo's commit ID to change.

The input data listed above (URLs, target filter, scanner) can be distributed for each tree-sitter language as a tarball and/or a docker image, and then used in CI or locally by whoever wants to get the official stats for their parser.

Recap of the suggested design:
* have a central repo where for each language we define the data and code needed to produce language stats.
* from the central repo, daily stats are produced and published as charts for known, public, tree-sitter projects.
* each tree-sitter project can run the same stats using their favorite method, such as performing a scan on each merge request.
